src/main/kotlin/uk/gov/dwp/dataworks/egress/services/impl/ManifestFileServiceImpl.kt

```diff
@@ -0,0 +1,210 @@
+                replace(TODAYS_DATE_PLACEHOLDER, todaysDate()))
+                logger.info("Got manifest target key",
+                    "specification" to "$specification",
+                    "target_key" to targetKey)
+
+                val (targetContents, putRequest) = targetContentsAndRequest(specification, targetKey)
+                logger.info("Got manifest target contents", "specification" to "$specification")
+
+                when (specification.transferType) {
+                    "S3" -> {
+
+                        logger.info("Transferring manifest file to s3",
+                            "specification" to "$specification",
+                            "target_key" to targetKey)
+
+                        egressClient(specification).putObject(putRequest, AsyncRequestBody.fromBytes(targetContents))
+                            .await()
+                        logger.info("Transferred manifest file to s3",
+                            "target_key" to targetKey,
+                            "specification" to "$specification")
+                        Pair(targetKey, true)
+                    }
+                    "SFT" -> {
+                        logger.info("Transferring manifest contents to file",
+                            "target_key" to targetKey,
+                            "specification" to "$specification")
+                        writeToFile(File(targetKey).name, specification.destinationPrefix, targetContents)
+                        logger.info("Transferred manifest contents to file",
+                            "target_key" to targetKey,
+                            "specification" to "$specification")
+                        Pair(targetKey, true)
+                    }
+                    else -> {
+                        logger.warn("Unsupported manifest type", "specification" to "$specification")
+                        Pair(targetKey, false)
+                    }
+                }
+            } else {
+                Pair(manifestFileName, false)
+            }
+        } catch (e: Exception) {
+            logger.error("Failed to egress manifest object", e, "specification" to "$specification")
+            Pair(manifestFileName, false)
+        }
+    }
+
+    private suspend fun targetContentsAndRequest(specification: EgressSpecification, targetKey:String): Pair<ByteArray, PutObjectRequest> {
+
+        val sourcePrefix = specification.sourcePrefix
+        val todaysDate = if (sourcePrefix.contains(TODAYS_YYYYMMDD_FORMATTED_DATE_PLACEHOLDER)) todaysDate("yyyyMMdd")
+                            else todaysDate()
+
+        // DataProductType only applicable for HTME Collections, or anything containing full or incremental in source prefix, last - is removed
+        val dataProductType = if (sourcePrefix.lowercase().contains("full")) "full"
```

Comment:
Can't we just set dataProductType to the value? i.e.
val dataProductType = sourcePrefix.lowercase()

Reply:
source prefix is full path to the location of that collection which contains full or incremental as substring